File: 2422-query-kth-smallest-trimmed-number/query-kth-smallest-trimmed-number.java
class Solution {
    class Pair{
        String ans;
        int index;
        Pair(String s,int i){
            this.ans=s;
            this.index=i;
        }
    }
    public int[] smallestTrimmedNumbers(String[] nums, int[][] queries) {
        List<Integer> li=new ArrayList<>();
        for(int []it :queries){
            int t=it[0];
            int y=it[1];
             List<Pair> new1=new ArrayList<>();
             for(int i=0;i<nums.length;i++){
                String num=nums[i];
             new1.add(new Pair(num.substring(num.length()-y),i));
             }
             Collections.sort(new1,(a,b)->{
                int cmp = a.ans.compareTo(b.ans);  
            if (cmp == 0) return a.index - b.index;    
             return cmp;
             });
            li.add(new1.get(t-1).index);
            new1.clear();
             
             
        }
        int t=queries.length;
        int [] arr=new int[li.size()];
        for(int i=0;i<t;i++){
            arr[i]=li.get(i);
        }
        return arr;
    }
}

File: 2422-query-kth-smallest-trimmed-number/query-kth-smallest-trimmed-number.cpp
class Solution {
public:
static bool compare(pair<string,int>&a,pair<string,int>&b){
    if(a.first==b.first)return a.second<b.second;
    return a.first<b.first;
}
    vector<int> smallestTrimmedNumbers(vector<string>& nums, vector<vector<int>>& queries) {
        int n=nums.size();
        vector<int>ans;
        for(auto & it :queries){
            int t=it[0];
            int y=it[1];
            vector<pair<string,int>>new1;
            for(int i=0;i<n;i++){
                string s=(nums[i]);
                new1.push_back({s.substr(s.size()-y),i});
            }
           sort(new1.begin(),new1.end(),compare);
           ans.push_back(new1[t-1].second);
           new1.clear();
        }
       
        return ans;
       

        
    }
};